src/UXClient/Utils.ts

```diff
@@ -1003,12 +1003,13 @@ class Utils {
             return data;
         }
 
-        let splittedByHit = str.split('<hit>');
+        let splitByTag = options && options.splitByTag ? options.splitByTag : 'hit';
+        let splittedByHit = str.split(`<${splitByTag}>`);
         splittedByHit.forEach((s, i) => {
             if (i === 0) {
                 data = data.concat(splitByNullGuid(s));
             } else {
-                let splittedByHitClose = s.split('</hit>');
```

Comment:
wait but what if </hit> is present?  should we have the default value of splitByTag be "hit"??

Reply:
yes, so basically the str we pass to this function was under the assumption of having just hit inside. I just forgot that I use Utils.mark manually to add mark tag into the search strings in context menu. so str is assumed to be a hit string (having only one type of highlighting, either hit or mark in my case). Instead of using Utils.mark, I can create a new one like Utils.hit to add hit tags around search strings as in /search api response for the context menu

Reply:
test